src/libasr/asr_utils.h

```diff
@@ -1081,6 +1081,27 @@ static inline bool is_generic(ASR::ttype_t &x) {
     }
 }
 
+static inline bool is_generic_function(ASR::symbol_t *x) {
+    switch (x->type) {
+        case ASR::symbolType::Function: {
+            ASR::Function_t *func_sym = ASR::down_cast<ASR::Function_t>(x);
+            return func_sym->n_type_params > 0;
+        }
+        default: return false;
+    }
+}
```

Comment:
Do we need a switch for this? I think if-else will fit better as only `Function` is to be checked and for all other cases the result would be `false`.

Reply:
Got it. Modified it.